pkgs/development/libraries/libbsd/default.nix

```diff
@@ -1,28 +1,48 @@
 { lib
 , stdenv
-, fetchurl
+, fetchFromGitLab
 , autoreconfHook
 , libmd
 , gitUpdater
 }:
 
-stdenv.mkDerivation rec {
+# Run `./get-version` for the new value when bumping the Git revision.
+let gitVersion = "0.11.7-55-g73b2"; in
+
+stdenv.mkDerivation {
   pname = "libbsd";
-  version = "0.11.7";
+  version = "unstable-2023-04-29";
 
-  src = fetchurl {
-    url = "https://libbsd.freedesktop.org/releases/${pname}-${version}.tar.xz";
-    hash = "sha256-m6oYYFnrvyXAYwjp+ZH9ox9xg8DySTGCbYOqar2KAmE=";
+  src = fetchFromGitLab {
+    domain = "gitlab.freedesktop.org";
+    owner = "libbsd";
+    repo = "libbsd";
+    rev = "73b25a8f871b3a20f6ff76679358540f95d7dbfd";
+    hash = "sha256-LS28taIMjRCl6xqg75eYOIrTDl8PzSa+OvrdiEOP1+U=";
   };
 
   outputs = [ "out" "dev" "man" ];
 
-  # darwin changes configure.ac which means we need to regenerate
-  # the configure scripts
+  enableParallelBuilding = true;
+
+  doCheck = true;
+
   nativeBuildInputs = [ autoreconfHook ];
   propagatedBuildInputs = [ libmd ];
 
-  patches = [ ./darwin.patch ];
+  patches = [
+    # Temporary fix to make `src/progname.c` compile on Darwin
+    ./darwin-fix-getprogname.patch
```

Comment:
Any specific reason this is not inside the darwin optional?

Reply:
It should be correct for all platforms (unlike the macOS-specific hack in the other patch), and on Linux it's a no-op, so I lean towards avoiding source code divergence between platforms where possible (e.g., if I think it would be reasonable for it to be committed upstream theres no reason to make it conditional; the only reason I'm not PRing this is that apparently there is a bunch of macOS porting work in the pipeline for `libbsd` already so it's likely redundant).

Arguably that means `darwin-fix-getprogname.patch` is a misleading name but I couldn't think of a better one (`fix-getprogname-on-platforms-with-__progname-support.patch` is a bit long...).